.github/workflows/publish.yml

```diff
@@ -0,0 +1,50 @@
+name: Release
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to PyPI
+    runs-on: ubuntu-latest
+    if: github.repository == 'astropy/pytest-astropy-header'
+
+    steps:
+    - uses: actions/checkout@master
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+
+    - name: Install python-build and twine
+      run: python -m pip install build "twine>=3.3"
+
+    - name: Build package
+      run: python -m build --sdist --wheel .
+
+    - name: List result
+      run: ls -l dist
+
+    - name: Check wheel
+      run: python -m twine check --strict dist/*
+
+    - name: Test package
+      run: |
+        cd ..
+        python -m venv testenv
+        testenv/bin/pip install astropy pytest pytest-astropy-header/dist/*.whl
+        testenv/bin/pytest pytest-astropy-header/tests
+
+    - name: Publish distribution 📦 to Test PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.test_pypi_password }}
+        repository_url: https://test.pypi.org/legacy/
+
+    - name: Publish distribution 📦 to PyPI
+      if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags')
```

Comment:
Eh... wait a minute... why is the PR event not skipping this part?

Reply:
The log is trying to push to PyPI test server too. That makes no sense.

Reply:
It's failing on the upload to TestPYPI above, not sure why.

Reply:
I wonder if they broke something on `master`...

Reply:
But then again, it should have been skipped in the first place. Why is the `if` logic not working?

Reply:
I found the cause and fixed it.

Reply:
Oh, what was the issue ?

Reply:
See 8d0667e 😆

Reply:
Yes, I was trying to publish on TestPyPI to test this, and I don't know why it was failing.

Reply:
The error said token doesn't have `pypi-` prefix. 🤷 

Anyway, since we agree to just "do it live" during the next release, there is no reason to dwell on it. 😸